views/Query/Collapsed.jsx

```diff
@@ -97,20 +100,28 @@ export default React.createClass({
           from: this.state.startDate.format('YYYYMMDD'),
           to: this.state.endDate.format('YYYYMMDD'),
         },
+        restrictedDateRange: this.state.restrictedDateRange,
       });
     }
   },
   dateButtonChanged(e) {
     var newDates = {};
+    var largestValue = "lasttwomonths";
+    var restrictedDateRange;
     let newButtonState = this.state.dateButtons.map((item) => {
       item.selected = item.value == e.target.value ? true : false;
       if (item.selected) {
         newDates = {startDate: item.startDate, endDate: item.endDate};
+        restrictedDateRange = item.value != largestValue;
       }
       return item;
     });
-    this.setState({dateButtons: newButtonState});
-    this.onDatesChange(newDates);
+    Promise.resolve(
+      this.setState({restrictedDateRange: restrictedDateRange})
+    ).then(() => {
```

Comment:
Because without it, the local state didn't update in time for onDatesChange to see the updated value of restrictedDateRange, and it remained false (or true, if it was true) until you clicked on the button a second time. You could see this happen if you comment out the promise lines and watch the query state as you click around.

There's a similar bug with the dates where sometimes it doesn't update the dates on the first click because the state update doesn't happen fast enough, which I have just realized. I'll fix that in another commit :D